modules/serverpod_auth/serverpod_auth_server/test/password_hash/argon2id_password_hash_test.dart

```diff
@@ -0,0 +1,228 @@
+import 'dart:convert';
+
+import 'package:serverpod_auth_server/src/business/password_hash.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Given password and fixed salt', () {
+    group('when generating password hash', () {
+      test('then hash has 4 sections split by \$.', () {
+        var passwordHash = PasswordHash.argon2id('hunter2', salt: 'saltySalt');
+
+        var parts = passwordHash.split('\$');
+        expect(parts, hasLength(4));
+      });
+
+      test('then second section contains algorithm name.', () {
+        var passwordHash = PasswordHash.argon2id('hunter2', salt: 'saltySalt');
+
+        var parts = passwordHash.split('\$');
+        expect(parts[1], 'argon2id');
+      });
+
```

Comment:
There are no requirements for the first section. It is probably empty since the string starts with `$` according to the convention. But it doesn't matter for us, we just ignore it.